### PR TITLE
Dashboard 3.9

### LINKED ISSUE #190: Update Dashboards to the latest ember version
Update Dashboards to the latest ember version

### PR DESCRIPTION
**Pull request by jkusa**
Resolves #190 

## Description

Ember 3.9 Navi Dashboards Upgrade

## Proposed Changes
- ember-cli-update changes and codemods
- use ember-promise-helpers instead of promise proxy properties
- normalize arrays in line-chart

### CODE REVIEW

File: packages/dashboards/addon/routes/dashboards/dashboard/widgets/new.js [hunk 1]
```diff
@@ -3,14 +3,14 @@
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  */
 
-import Ember from 'ember';
+import { A } from '@ember/array';
+
```
Comment (cythrawll) [inline packages/dashboards/addon/routes/dashboards/dashboard/widgets/new.js]: whitespace?

File: packages/dashboards/addon/services/dashboard-data.js [hunk 2]
```diff
@@ -2,31 +2,37 @@
  * Copyright 2017, Yahoo Holdings Inc.
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  */
-import Ember from 'ember';
+import { isEmpty } from '@ember/utils';
+
```
Comment (cythrawll) [inline packages/dashboards/addon/services/dashboard-data.js]: whitespace?

File: packages/dashboards/tests/unit/helpers/get-widget-test.js [hunk 3]
```diff
@@ -1,28 +1,22 @@
...
-  });
+    run(() => {
+      store.push({ data: { id: 2, type: 'dashboard-widget' } });
+    });
 
-  assert.equal(getWidget.compute([2]).get('id'), '2', 'widget with given id is returned');
+    assert.equal(getWidget.compute([2]).get('id'), '2', 'widget with given id is returned');
 
-  assert.equal(getWidget.compute([24]), null, 'undefined is returned when no widget has that id');
+    assert.equal(getWidget.compute([24]), null, 'undefined is returned when no widget has that id');
```
Comment (cythrawll) [inline packages/dashboards/tests/unit/helpers/get-widget-test.js]: `null is returned`

File: packages/dashboards/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# How To Contribute
```
Comment (Alex-Aralis) [inline packages/dashboards/CONTRIBUTING.md]: We already have a contrib file on the top level.

File: packages/dashboards/tests/acceptance/dashboards-test.js [hunk 5]
```diff
@@ -396,7 +400,7 @@ module('Acceptance | Dashboards', function(hooks) {
     await visit('/dashboards/1');
     await click('.editable-label__icon');
     await fillIn('.editable-label__input', 'A new title');
-    await triggerEvent('.editable-label__input', 'blur');
+    await await blur('.editable-label__input');
```
Comment (mattgraham1995) [inline packages/dashboards/tests/acceptance/dashboards-test.js]: double await

**Comment by cythrawll** [comment]
package lock has ouroath.

File: packages/dashboards/CONTRIBUTING.md (hunk 4 above)
Comment (jkusa) [reply packages/dashboards/CONTRIBUTING.md]: Adding to the local `.gitignore`

File: packages/dashboards/tests/acceptance/dashboards-test.js (hunk 5 above)
Comment (jkusa) [reply packages/dashboards/tests/acceptance/dashboards-test.js]: wow, too many codemods

**Pull request merged**